test/fw/ptl/utils/plugins/ptl_test_runner.py

```diff
@@ -706,55 +707,89 @@ def __are_requirements_matching(param_dic=None, test=None):
             param_count['num_' + key] = len(param_dic[key])
         for pk in param_count:
             if param_count[pk] < eff_tc_req[pk]:
-                return False
+                _msg = 'available ' + pk + " is less than requirement"
+                logger.error(_msg)
+                return _msg
         for hostname in param_dic['moms']:
             si = SystemInfo()
             si.get_system_info(hostname)
             available_sys_ram = getattr(si, 'system_ram', None)
             if available_sys_ram is None:
-                return False
+                _msg = 'failed to get ram info on host: ' + hostname
+                logger.error(_msg)
+                return _msg
             elif eff_tc_req['min_mom_ram'] >= available_sys_ram:
-                return False
+                _msg = 'avail ram is less than test requirement on host: '
+                _msg += hostname
+                logger.error(_msg)
+                return _msg
             available_sys_disk = getattr(si, 'system_disk', None)
             if available_sys_disk is None:
-                return False
+                _msg = 'failed to get disk info on host: ' + hostname
+                logger.error(_msg)
+                return _msg
             elif eff_tc_req['min_mom_disk'] >= available_sys_disk:
-                return False
+                _msg = 'avail disk is less than test requirement on host: '
+                _msg += hostname
+                logger.error(_msg)
+                return _msg
         for hostname in param_dic['servers']:
             si = SystemInfo()
             si.get_system_info(hostname)
             available_sys_ram = getattr(si, 'system_ram', None)
             if available_sys_ram is None:
-                return False
+                _msg = 'failed to get ram info on host: ' + hostname
+                logger.error(_msg)
+                return _msg
             elif eff_tc_req['min_server_ram'] >= available_sys_ram:
-                return False
+                _msg = 'avail ram is less than requirement on host: '
+                _msg += hostname
+                logger.error(_msg)
+                return _msg
             available_sys_disk = getattr(si, 'system_disk', None)
             if available_sys_disk is None:
-                return False
+                _msg = 'failed to get disk info on host: ' + hostname
+                logger.error(_msg)
+                return _msg
             elif eff_tc_req['min_server_disk'] >= available_sys_disk:
-                return False
+                _msg = 'avail disk is less than requirement on host: '
+                _msg += hostname
+                logger.error(_msg)
+                return _msg
         if set(param_dic['moms']) & set(param_dic['servers']):
             if eff_tc_req['no_mom_on_server']:
-                return False
+                _msg = 'no mom on server: '
```

Comment:
the colon and space indicates, we will log something after colon, but we are not doing that.

Reply:
@prakashcv13  content of /proc/meminfo file
centos6- 
MemTotal:        3974464 kB
MemFree:         1471012 kB
Buffers:          662096 kB
Cached:          1357724 kB
SwapCached:            0 kB

centos7-
MemTotal:        7994076 kB
MemFree:          255032 kB
MemAvailable:    3217732 kB
Buffers:          263300 kB
Cached:          2849136 kB
SwapCached:         1008 kB
so, I check if MemAvailable is present in first three lines then i take MemAvailable as system_ram otherwise i take MemFree as system_ram. reason is after first 3 lines there  is not possibility of getting MemAvailable.

Reply:
get_MemAvailable = False
            ram_search_count = 0
            for i in mem_info['out']:
                ram_search_count += 1
                if "MemTotal" in i:
                    self.system_total_ram = float(i.split()[1]) / (2**20)
                elif "MemFree" in i:
                    MemFree = float(i.split()[1]) / (2**20)
                elif "MemAvailable" in i:
                    get_MemAvailable = True
                    MemAvailable = float(i.split()[1]) / (2**20)
                if get_MemAvailable:
                    self.system_ram = MemAvailable
                    break
                elif ram_search_count >= 3:
                    self.system_ram = MemFree
                    break